### DISCUSSION TITLE
Frame limiting

### DISCUSSION THREAD

**Discussion by waylaa** [dotnet/Silk.NET#2229] [Q&A]
I am trying to make a very simple video player in OpenGL (and Vulkan afterwards). I decode each frame using the FFmpeg.Autogen library and send the pixel data as a ```byte*``` and the frame dimensions to a queue in my OpenGL renderer class. Each frame gets dequeued in the ```Render``` event where a texture gets updated with the new pixel data. The issue is that the playback is extremely fast. I have tried using ```Thread.Sleep```, setting the ```IWindow```'s properties ```FramesPerSecond``` and ```UpdatesPerSecond``` to 30 or 60, experimenting with delta time and apparently nothing works. Also, no code exists that demonstrates how to implement a frame limiter in the examples folder.

**Comment by Perksey** [comment]
I would make sure VSync is off to frame limit on the window. But to be honest, for your application the approach I would use would be using a cumulative delta time to determine when to switch to the next frame. e.g. `if ((_cumDelta += deltaTime) > (1 / 60)) { AdvanceFrame(numberOfFrames: (int)(_cumDelta / (1 / 60))); _cumDelta %= (1 / 60); }` (where the video FPS is 60 FPS) in your render loop. Frame limiting the window to the video FPS could cause tearing for videos that have a higher FPS than the monitor, so you will want these to be independent ideally.

**Reply by waylaa** [reply]
I used your example as reference and I made it work. Do you mind sharing an example with the framerate and speed being independent of each other, so for example, you can have 1000fps while the video is being rendered at 60fps without the video slowing down or speeding up because of the frame rate.
